library/ssl_misc.h

```diff
@@ -1115,6 +1115,59 @@ void mbedtls_ssl_handshake_wrapup( mbedtls_ssl_context *ssl );
 void mbedtls_ssl_handshake_wrapup_tls13( mbedtls_ssl_context *ssl );
 #endif
 
+/**
+ * To let TLS1.2 and TLS1.3 work together, we divide the task into two steps.
+ * Below inline functions are required by first step.
+ * 1. Configured as TLS1.2 or TLS1.3. Identify the protocol version based on configuration field.
+ * 2. Configured as TLS1.2 and TLS1.3. Identify the protocal version based on runtime parameter.
+ */
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+static inline int is_configured_as_tls1_3(const mbedtls_ssl_config *conf)
+{
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2)
+    if(conf->max_major_ver==MBEDTLS_SSL_MAJOR_VERSION_3
+        && conf->min_major_ver == MBEDTLS_SSL_MAJOR_VERSION_3
+        && conf->max_minor_ver == MBEDTLS_SSL_MINOR_VERSION_4
+        && conf->min_minor_ver == MBEDTLS_SSL_MINOR_VERSION_4)
+        return( 1 );
+    return( 0 );
+#else /* defined(MBEDTLS_SSL_PROTO_TLS1_2) */
+    (( void )conf);
+    return( 1 );
+#endif /* defined(MBEDTLS_SSL_PROTO_TLS1_2) */
+}
+#endif /* defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL) */
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2)
+static inline int is_configured_as_tls1_2_or_earlier(const mbedtls_ssl_config *conf)
```

Comment:
`_or_earlier()` is no longer needed after the rebase - we no longer support TLS versions below 1.2.